### PR TITLE
Remove eager/lazy cache

### PR DESCRIPTION
**Pull request by rigazilla**
Eager cache needs to be removed from the cache map
... also lazy needs the same

### CODE REVIEW

File: manager/src/main/java/io/gingersnapproject/Caches.java [hunk 1]
```diff
@@ -176,6 +178,17 @@ public Uni<Boolean> remove(String name, String key) {
       return Uni.createFrom().item(Boolean.FALSE);
    }
 
+   public void removeCache(String ruleName) {
```
Comment (ryanemerson) [inline manager/src/main/java/io/gingersnapproject/Caches.java]: Can't we just do `@Observes RuleEvents.EagerRuleRemoved ev` here and avoid calling a method from `DatabaseHandler`?

Comment (rigazilla) [reply manager/src/main/java/io/gingersnapproject/Caches.java]: Yeah maybe yes, let me check that the exec order doesn't matter.

Comment (rigazilla) [reply manager/src/main/java/io/gingersnapproject/Caches.java]: seems safe, changed to use events

**Comment by ryanemerson** [comment]
Thanks @rigazilla.

I'll rebase https://github.com/gingersnap-project/cache-manager/pull/62 so that I can update the returned status codes.

**Pull request merged**